base_layer/wallet_ffi/src/lib.rs

```diff
@@ -322,49 +1246,161 @@ pub unsafe extern "C" fn comms_config_destroy(wc: *mut TariCommsConfig) {
 
 /// ------------------------------------- Wallet -------------------------------------------------///
 
+/// Creates a TariWallet
+///
+/// ## Arguments
+/// `config` - The TariCommsConfig pointer
+///
+/// ## Returns
+/// `*mut TariWallet` - Returns a pointer to a TariWallet, note that it returns ptr::null_mut()
+/// if config is null, a wallet error was encountered or if the runtime could not be created
 #[no_mangle]
 pub unsafe extern "C" fn wallet_create(config: *mut TariCommsConfig) -> *mut TariWallet {
-    // TODO Check that the config is not null, how do you deal with the case that it is null?
-
+    if config.is_null() {
+        return ptr::null_mut();
+    }
     // TODO Gracefully handle the case where these expects would fail
-    let runtime = Runtime::new().expect("Could not create a Tokio runtime.");
-    let w = TariWallet::new(
-        WalletConfig {
-            comms_config: (*config).clone(),
+    let runtime = Runtime::new();
+    let w;
+    match runtime {
+        Ok(runtime) => {
+            w = TariWallet::new(
+                WalletConfig {
+                    comms_config: (*config).clone(),
+                },
+                WalletMemoryDatabase::new(),
+                runtime,
+            );
+            match w {
+                Ok(w) => Box::into_raw(Box::new(w)),
+                Err(_) => ptr::null_mut(),
+            }
         },
-        WalletMemoryDatabase::new(),
-        runtime,
-    )
-    .expect("Could not create wallet"); // expect needs to change due to it panicking.
-    Box::into_raw(Box::new(w))
+        Err(_) => ptr::null_mut(),
+    }
 }
 
+/// Generates test data
```

Comment:
This is not a particularly helpful docstring as it just repeats the name of the function. I would say provide at least a sentence describing the actual behaviour. You can copy it from the long docstring I wrote for the top of this file.

This goes for many of the functions in this PR. I wouldnt bother with extra detail on destructuring methods but on any methods that have non0trivial behaviour describe what the function does and in what context it would be used.

Reply:
Have expanded the documentation further, it possibly will need more work.

Have also expanded the test coverage further.